packages/core/src/indexing/postgres/__tests__/postgres-index-api.test.ts

```diff
@@ -0,0 +1,158 @@
+import { jest } from '@jest/globals'
+import { PostgresIndexApi } from '../postgres-index-api.js'
+import { StreamID } from '@ceramicnetwork/streamid'
+import knex, { Knex } from 'knex'
+import pgSetup from '@databases/pg-test/jest/globalSetup'
+import pgTeardown from '@databases/pg-test/jest/globalTeardown'
+
+const STREAM_ID_A = 'kjzl6cwe1jw145m7jxh4jpa6iw1ps3jcjordpo81e0w04krcpz8knxvg5ygiabd'
+const STREAM_ID_B = 'kjzl6cwe1jw147dvq16zluojmraqvwdmbh61dx9e0c59i344lcrsgqfohexp60s'
+const CONTROLLER = 'did:key:foo'
+
+let dbConnection: Knex
+jest.setTimeout(150000) // 2.5mins timeout for initial docker fetch+init
+
+beforeAll(async () => {
+  await pgSetup()
+})
+
+beforeEach(async () => {
+  dbConnection = knex({
+    client: 'pg',
+    connection: process.env.DATABASE_URL,
+  })
+})
+
+afterEach(async () => {
+  await dropMidTables()
+  await dbConnection.destroy()
+})
+
+export async function listMidTables() {
+  const dataArr = []
+  await dbConnection
+    .select('tablename')
+    .from('pg_tables')
+    .whereRaw("schemaname='public' AND tablename LIKE ('kjz%')")
+    .then(function (result) {
```

Comment:
@willex Now it is all right, but going forward, please, lean towards either async/await or `.then` style. async/await is preferred.